### PR TITLE
Add basic note parsing

### PR DESCRIPTION
**Pull request by soul2197**
@JPricey 

Depends on https://github.com/soul2197/midi.js/pull/24

### CODE REVIEW

File: src/parser.js [hunk 1]
```diff
@@ -70,7 +71,7 @@ module.exports = class {
           throw 'F0 Sysex Event unsupported';
         case 0xF7: // F7 Sysex Event
           throw 'F7 Sysex Event unsupported';
-        case 0xFF: // Meta Event
+        case 0xFF: { // Meta Event
```
Comment (JPricey) [inline src/parser.js]: Why use curly braces?

File: src/parser.js [hunk 2]
```diff
@@ -101,31 +103,111 @@ module.exports = class {
...
+          let channel = event_type & 0x0F;
+
+          switch (channel_event_type) {
+            case 0x80: { // Note off
+              let key = chunk.data[index + delta_time_len + 1];
+              let velocity = chunk.data[index + delta_time_len + 2];
+
+              track.events.push({
+                delta_time: delta_time,
+                type: 'note_off',
```
Comment (JPricey) [inline src/parser.js]: If the type is just to help with debugging this is fine, but if we check the type a lot during runtime, I prefer to use "enums" (well, named consts is the closest you can get in JS...). Need that sweet sweet performance.

File: src/parser.js [hunk 3]
```diff
@@ -101,31 +103,111 @@ module.exports = class {
...
-          // TODO: handle these properly
-          file.tracks.push(track);
-          return;
+        }
+        default: {
+          let channel_event_type = event_type & 0xF0;
+          let channel = event_type & 0x0F;
+
+          switch (channel_event_type) {
+            case 0x80: { // Note off
```
Comment (JPricey) [inline src/parser.js]: Note on/off cases are almost the same code. I think if/else is better here than switch case so you can handle them both in the same block.

File: src/parser.js [hunk 4]
```diff
@@ -101,31 +103,111 @@ module.exports = class {
...
+            }
           }
           index += delta_time_len + event_length_len + event_length + 2;
           break;
-        default:
-          // TODO: handle these properly
-          file.tracks.push(track);
-          return;
+        }
+        default: {
```
Comment (JPricey) [inline src/parser.js]: Misuse of default imo but the alternative is messy too... I guess this is fine i'm just not too happy about it.

File: src/parser.js [hunk 5]
```diff
@@ -101,31 +103,111 @@ module.exports = class {
...
+              // TODO: set active instrument to program_number?
+              index += delta_time_len + 2;
+              break;
+            }
+            case 0xD0: // Channel Key Pressure
+              throw 'Channel Key Pressure unsupported';
+            case 0xE0: // Pitch Bend
+              throw 'Pitch Bend unsupported';
+            default: // Running Status
+              throw('Running Status unsupported');
```
Comment (JPricey) [inline src/parser.js]: inconsistent parens use

File: src/parser.js (hunk 2 above)
Comment (soul2197) [inline src/parser.js]: going to make another PR to add `"enums"`

File: src/parser.js (hunk 3 above)
Comment (soul2197) [inline src/parser.js]: will refactor in the enum branch
